utils/oomparser/oomparser.go

```diff
@@ -101,132 +90,46 @@ func checkIfStartOfOomMessages(line string) bool {
 	return false
 }
 
-// reads the file and sends only complete lines over a channel to analyzeLines.
-// Should prevent EOF errors that occur when lines are read before being fully
-// written to the log. It reads line by line splitting on
-// the "\n" character.
-func readLinesFromFile(lineChannel chan string, ioreader *bufio.Reader) error {
-	linefragment := ""
-	var line string
-	var err error
-	for true {
-		line, err = ioreader.ReadString('\n')
-		if err != nil && err != io.EOF {
-			glog.Errorf("exiting analyzeLinesHelper with error %v", err)
-			close(lineChannel)
-			break
-		}
-		if line == "" {
-			time.Sleep(100 * time.Millisecond)
-			continue
-		}
-		if err == nil {
-			lineChannel <- linefragment + line
-			linefragment = ""
-		} else { // err == io.EOF
-			linefragment += line
-		}
-	}
-	return err
-}
-
-// Calls goroutine for readLinesFromFile, which feeds it complete lines.
-// Lines are checked against a regexp to check for the pid, process name, etc.
-// At the end of an oom message group, StreamOoms adds the new oomInstance to
-// oomLog
-func (self *OomParser) StreamOoms(outStream chan *OomInstance) {
-	lineChannel := make(chan string, 10)
-	go func() {
-		readLinesFromFile(lineChannel, self.ioreader)
-	}()
+// StreamOoms writes to a provided a stream of OomInstance objects representing
+// OOM events that are found in the logs.
+// It will blocks and should be called from a goroutine.
+func (self *OomParser) StreamOoms(outStream chan<- *OomInstance) {
+	kmsgEntries := self.parser.Parse()
```

Comment:
Question: Should we call `self.parser.SeekEnd()` here or not?

I think `SeekEnd` more closely matches the `journalctl .. -f` behavior of before (start at the end of kmsg), but I think that looking a little into the past makes the kubelet behave more correctly after a restart.

Will OOM events that happen while the kubelet is down, but are detected on kubelet startup actually do the right thing?
Are they idempotent so that power-cycling the kubelet having bunches of duplicate events is okay?